src/main/java/io/github/debuggyteam/architecture_extensions/api/ArchExIntegration.java

```diff
@@ -1,20 +1,39 @@
 package io.github.debuggyteam.architecture_extensions.api;
 
-import java.util.function.Consumer;
-
 import org.jetbrains.annotations.ApiStatus;
 
-import net.minecraft.item.ItemGroup;
+import net.minecraft.block.Block;
 
 public interface ArchExIntegration {
 	final String ENTRYPOINT_KEY = "architecture_extensions";
 
+	/**
+	 * Called by Architecture Extensions when it's ready to create blocks on behalf of a mod.
+	 * @param ctx A Context object which can honor block creation requests
+	 */
 	void integrate(Context ctx);
+	
+	/**
+	 * Called by Architecture Extensions when it creates a block that was requested by this integration class. You can
+	 * use this function to add the new block to an ItemGroup, generate additional data, or emit diagnostic information.
```

Comment:
I think putting a newline between the method description and the params would make it easier to read the javadocs.